xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/objects/classes/DBListClass.java

```diff
@@ -282,8 +285,14 @@ public String getQuery(XWikiContext context)
                     select.append("doc." + idField);
                 } else {
                     select.append("idprop.value");
-                    fromStatements.add("StringProperty as idprop");
                     whereStatements.add("obj.id=idprop.id.id and idprop.id.name='" + idField + "'");
+                    // Get the from statements according to the type of property
+                    try {
+                        String classType = getPropertyType(classname, idField,context);
+                        fromStatements.add(classType + " as idprop");
+                    } catch (Exception e) {
+                        LOGGER.error("Failed to parse SQL script [{}]. Continuing with non-rendered script.", sql, e);
```

Comment:
It seems you copied this line from some other place in this file.. but the context here is quite different.